python/langsmith/_internal/_background_thread.py

```diff
@@ -88,6 +94,42 @@ def _tracing_thread_drain_queue(
     return next_batch
 
 
+def _tracing_thread_drain_compressed_buffer(
+    client: Client,
+    size_limit: int = 100,
+    size_limit_bytes: int = 50 * 1024 * 1024
+) -> Optional[Iterable[bytes]]:
+    with client._buffer_lock:
+        current_size = client.compressed_runs_buffer.tell()
+
+        # Check if we should send now
+        if not (client._run_count >= size_limit or current_size >= size_limit_bytes):
+            return None
+
+        # Write final boundary and close compression stream
+        client.compressor_writer.write(f'--{client.boundary}--\r\n'.encode())
+        client.compressor_writer.flush()
+        client.compressor_writer.close()
```

Comment:
I think you just need to call `close()` here

Reply:
see above comment